design-documents/graph-ql/coverage/gift-registry.graphqls

```diff
@@ -0,0 +1,278 @@
+type Customer {
+    gitft_registry_list: [GiftRegistry]
+    gitft_registry(id: ID!): GiftRegistry
+}
+
+type Query {
+    giftRegistryTypes: [GiftRegistryType] @doc(description: "Get a list of available gift registry types")
+    giftRegistrySearch(
+        registrant_firstname: String!,
+        registrant_lastname: String!,
+        gift_registry_type_id: ID,
+        searchable_dynamic_attributes: [GiftRegistryDynamicAttributeInput] @doc(description: "For select attributes ID should be provided expected. For range search, '_from' and '_to' suffixes can be added to the attribute code. For text attributes provide value for exact matching.")
+    ): [GiftRegistry] @doc(description: "Gift registry search by registrant name and additional searchable attributes.")
+}
+
+type Mutation {
+    # All mutations below should only be accessible to the registry owner
```

Comment:
> All mutations below should only be accessible to the registry owner

Shouldn't they be accessible to everyone, but throw for folks without correct authorization?

Reply:
Exactly, they should give authorization errors, that's what I meant here. We don't have a way to hide fields from specific users.

Reply:
Clarified the comment.